src/extension/tab/hook.ts

```diff
@@ -50,7 +50,9 @@ type Hook = {
 const DEVTOOLS_KEY = Symbol.for("apollo.devtools");
 
 const tab = createWindowActor(window);
-const messageAdapter = createWindowMessageAdapter(window);
+const messageAdapter = createWindowMessageAdapter(window, {
+  jsonSerialize: true,
```

Comment:
We only need to do this for the data sent from the client into the extension since we control the payloads from the extension to this script and can better guarantee we're always sending serializable values.